lib/dolly/mango_index.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'dolly/document'
+
+module Dolly
+  class MangoIndex
+    class << self
+      DESIGN = '_index'
+
+      def all
+        Dolly::Document.connection.get(DESIGN)[:indexes]
+      end
+
+      def create(name, fields, type = 'json')
+        Dolly::Document.connection.post(DESIGN, build_index_structure(name, fields, type))
+      end
+
+      def find_by_fields(fields)
+        all.find do |index_doc|
+          index_doc.dig(:def, :fields).map(&:keys).flatten == fields
```

Comment:
dig can return `nil` is that ok?

Reply:
Yeah it's fine, the fields wont match and `find` will not return that index document